src/main/perf/SearchTask.java

```diff
@@ -294,23 +294,14 @@ public void go(IndexState state) throws IOException {
         if (vectorField != null) {
           IndexReader reader = searcher.getIndexReader();
           List<LeafReaderContext> leaves = reader.leaves();
-          List<RandomAccessVectorValues> perLeafVectors = new ArrayList<>();
-          for (LeafReaderContext ctx : leaves) {
-            VectorValues vectorValues = ctx.reader().getVectorValues(vectorField);
-            if (vectorValues != null) {
-              perLeafVectors.add(((RandomAccessVectorValuesProducer) vectorValues).randomAccess());
-            } else {
-              perLeafVectors.add(null);
-            }
-          }
           for (int i = 0; i < hits.scoreDocs.length; i++) {
             ScoreDoc scoreDoc = hits.scoreDocs[i];
             int doc = scoreDoc.doc;
             int segment = ReaderUtil.subIndex(doc, leaves);
-            RandomAccessVectorValues vectors = perLeafVectors.get(segment);
-            if (vectors != null) {
-              vectors.vectorValue(doc - leaves.get(segment).docBase);
-            }
+            VectorValues vv = leaves.get(segment).reader()
+                                                .getVectorValues(vectorField);
```

Comment:
It's somewhat costly to allocate a new `VectorValues` for each document, when we only need one per leaf. I guess we may not notice if the result count is small relative to the leaf count, but ... could we restore the loop above replacing the `List<RandomAccessVectorValues> perLeafVectors` with `List<VectorValues> perLeafVectors`?

Reply:
Updated to restore the loop. I somehow convinced myself that the overhead for a new `VectorValue` object is minimal(just some pointer updates). But looks like thats not the case.